src/ucp/core/ucp_ep.h

```diff
@@ -885,4 +887,26 @@ ucs_status_t ucp_ep_query_sockaddr(ucp_ep_h ucp_ep, ucp_ep_attr_t *attr);
  */
 ucs_status_t ucp_ep_realloc_lanes(ucp_ep_h ep, unsigned new_num_lanes);
 
+/**
+ * @brief Callback to notify about ep restart completion.
+ *
+ * @param [in]  ep       Endpoint which completed restart operation.
+ * @param [in]  arg      User argument to be passed to the callback.
+ */
+typedef void (*ucp_ep_restart_completion_cb_t)(ucp_ep_h ep, void *arg);
+
+/**
+ * @brief Restarts all pending requests by selecting a new protocol for each
+ *        one and reset old protocol.
+ *
+ * @param [in] ep   Endpoint object.
+ * @param [in] cb   Callback to be called upon completion.
+ * @param [in] arg  Argument to be passed to callback.
+ *
+ * @return Error code as defined by @ref ucs_status_t
+ */
+ucs_status_t ucp_ep_pending_schedule_restart(ucp_ep_h ep,
```

Comment:
Where is this function called besides tests? The PR seems incomplete...

Reply:
it is used in RC/DC while reconfiguring EP

Reply:
Do you mean other PR? pls add a reference

Reply:
it's WIP, will update once I add it.

Reply:
are cb/arg indeed required to be parameterized?

Reply:
actually it seems that A ucp_ep API is not needed (because of the usage in RC/DC).
so will remove it alltogether

Reply:
so, not addressed?

Reply:
done